genie-agent/src/main/java/com/netflix/genie/agent/configs/FetchingCacheServiceCleanUpTaskExecutorConfig.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.configs;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+/**
+ * Configuration for {@link com.netflix.genie.agent.execution.services.FetchingCacheService} clean up task.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Configuration
+public class FetchingCacheServiceCleanUpTaskExecutorConfig {
+
+    /**
+     * Get a task executor for cleaning up {@link com.netflix.genie.agent.execution.services.FetchingCacheService}.
+     *
+     * @param poolSize The number of threads desired for this system
+     * @return The task executor to use
+     */
+    @Bean
+    public TaskExecutor fetchingCacheServiceCleanUpTaskExecutor(
+        @Value("${genie.fetching.cache.service.cleanup.executor.pool.size:1}") final int poolSize
```

Comment:
You should document this new property in `_properties.adoc`.
The name could be a little more user-friendly too, maybe `genie.agent.cache.cleanup.executorPoolSize`?

Reply:
Actually, thinking about this a little more.
This is not a server configuration, it's an agent runtime configuration.
There are many other places in the agent where we temporarily hardcoded some configuration value and sometimes later we'll need to come up with a strategy to make those configurable.
Some will become command-line options, some will be pushed down by the server, some we may not expose at all.
So could you leave a TODO here for a later pass of client configurations, and temporarily hardcode `1`?

Reply:
done